ppa/archive/templatetags/ppa_tags.py

```diff
@@ -113,3 +115,29 @@ def esc(x):
             ]
         )
     )
+
+
+@register.filter
+def first_page(page_range):
+    """Extract first page number from a page range like '25-30' or '25'"""
+    if not page_range:
+        return ""
+    return page_range.split("-")[0].strip()
+
+
+@register.filter
+def last_page(page_range):
+    """Extract last page number from a page range like '25-30' or '25'"""
+    if not page_range:
+        return ""
+    parts = page_range.split("-")
+    return parts[-1].strip() if len(parts) > 1 else parts[0].strip()
+
+
+@register.simple_tag(takes_context=True)
+def absolute_url(context, relative_url):
```

Comment:
We already have [django-fullurl](https://github.com/Flimm/django-fullurl) as a dependency (used for a similar need in the open search header), so let's use the `fullurl` template tag instead of adding our own.